airflow/utils/log/logging_mixin.py

```diff
@@ -218,24 +224,14 @@ def set_context(logger, value):
     :param logger: logger
     :param value: value to set
     """
-    while logger:
-        orig_propagate = logger.propagate
+    if value is not None:
```

Comment:
The changes in this function are mainly what this PR is about.  We no longer need to "walk" the logger hierarchy because we no longer duplicate task logging handlers.

And as a result we no longer need to disable propagation by default, so we can revert back to the prior approach of just disabling when explicitly asked to do so.